Sources/Build/BuildPlan.swift

```diff
@@ -542,16 +553,28 @@ public final class SwiftTargetBuildDescription {
     /// These are the source files derived from plugins.
     private var pluginDerivedSources: Sources
 
+    /// These are the resource files derived from plugins.
+    private var pluginDerivedResources: [Resource]
+
     /// Path to the bundle generated for this module (if any).
     var bundlePath: AbsolutePath? {
-        buildParameters.bundlePath(for: target)
+        if let bundleName = target.underlyingTarget.potentialBundleName, !resources.isEmpty {
+            return buildParameters.bundlePath(named: bundleName)
+        } else {
+            return .none
+        }
     }
 
     /// The list of all source files in the target, including the derived ones.
     public var sources: [AbsolutePath] {
         target.sources.paths + derivedSources.paths + pluginDerivedSources.paths
     }
 
+    /// The list of all resource files in the target, including the derived ones.
+    public var resources: [Resource] {
+        target.underlyingTarget.resources + pluginDerivedResources
```

Comment:
should we add `target.resources` pointing to `target.underlyingTarget.resources`?

Reply:
Maybe, but reaching into `target.underlyingTarget` is a common pattern, so I think if we want to change that, that should be a separate PR.

Reply:
probably worth a dedicated cleanup then. should we add a ticket to track?

Reply:
rdar://92180880 (Clean up use of `target.underlyingTarget`)